### PR TITLE
chore: `settings/teams` and `settings/billing` - remove pages router and use app router

### PR DESCRIPTION
**Pull request by hbjORbj**
## What does this PR do?

- Fixes CAL-4383
- remove pages router and use app router for all pages in `settings/teams`
- remove env vars not needed anymore as a result

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A - I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

- all settings/teams/**/* pages must work as expected

### CODE REVIEW

File: yarn.lock
Comment (sean-brydon) [inline yarn.lock:1]: Can we restore this file?

**Review by sean-brydon — DISMISSED** [review]
LGTM but lets add yarn.lock back in from main

File: yarn.lock
Comment (hbjORbj) [reply yarn.lock:1]: Yeah sure, done!

**Comment by hbjORbj** [comment]
TODO: Test on Prod

**Comment by hbjORbj** [comment]
Added booking liimits page as a follow up of #16614

**Comment by hbjORbj** [comment]
Blocked by the failing integration test. Seems like there's a problem with the test.

**Review by joeauyeung — DISMISSED** [review]

**Review by joeauyeung — APPROVED** [review]

**Pull request merged**